File: 1768-merge-strings-alternately/1768-merge-strings-alternately.ts
function mergeAlternately(word1: string, word2: string): string {
    const merged: string[] = [];
    const minLength = Math.min(word1.length, word2.length);

    for (let i = 0; i < minLength; i++) {
        merged.push(word1[i], word2[i]);
    }

    if (word1.length > word2.length) {
        merged.push(...word1.slice(minLength));
    } else if (word2.length > word1.length) {
        merged.push(...word2.slice(minLength));
    }

    return merged.join("");
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        
        merged_string = []
        i, j = 0, 0
        # Loop through both strings while characters remain in both
        while i < len(word1) and j < len(word2):
            merged_string.append(word1[i])
            merged_string.append(word2[j])
            i += 1
            j += 1
        # Add remaining characters from the longer string (if any)
        if i < len(word1):
            merged_string.append(word1[i:])
        if j < len(word2):
            merged_string.append(word2[j:])
        
        return ''.join(merged_string)